### PR TITLE
Add routing on the panel to allow navigation of individual thread

### PR DESCRIPTION
**Pull request by jybao1**
## Changes:

- Previously, individual thread on the panel is not clickable. A route is defined in app-routing.module.ts to enable navigation into each thread when click on it.
- Separated out ``thread-list`` component and ``thread-detail`` component from ``panel`` component and replaced with ``<router-outlet>`` element. 
Note: Implementation for these two components are not included in this PR, because they involve major changes that are not relevant for the purpose of this PR. I will create another PR with them later. 

## Refactoring: 

- Refactored import in alphabetic order.
- Refactored folder structure: moved modal under panel folder.

### CODE REVIEW

File: ui/src/app/panel/panel.component.html [hunk 1]
```diff
@@ -7,81 +7,27 @@
...
-          <a class="tag is-rounded is-medium is-primary">tag</a>
-          <a class="tag is-rounded is-medium is-success">tag</a>
-          <a class="tag is-rounded is-medium is-warning">tag</a>
-          <a class="tag is-rounded is-medium is-danger">tag</a>
-          <a class="tag is-rounded is-medium is-link">tag</a>
-          <a class="tag is-rounded is-medium is-success">tag</a>
-          <a class="tag is-rounded is-medium is-info">tag</a>
-          <a class="tag is-rounded is-medium is-success">tag</a>
-          <a class="tag is-rounded is-medium is-dark">tag</a>
+          <a class="tag is-light is-medium is-primary">tag</a>
```
Comment (jybao1) [inline ui/src/app/panel/panel.component.html:10]: Modified tag color and shape

File: ui/src/app/app.module.ts [hunk 2]
```diff
@@ -31,6 +37,7 @@ import { ModalComponent } from './modal/modal.component';
     MatChipsModule,
     MatIconModule,
     IonicModule.forRoot(),
+    GoogleChartModule
```
Comment (jybao1) [inline ui/src/app/app.module.ts:40]: Will also create a separate PR for showing chart for a tag.

File: ui/src/app/service/data.service.ts [hunk 3]
```diff
@@ -2,15 +2,15 @@ import {Injectable} from '@angular/core';
 
 import {Observable, Subject} from 'rxjs';
 
-import {TagpostServiceClient} from 'compiled_proto/src/proto/Tagpost_rpcServiceClientPb';
+import {Tag, Thread} from '../../compiled_proto/src/proto/tagpost_pb';
```
Comment (chnlich) [inline ui/src/app/service/data.service.ts:5]: Suggest to use absolute path at import stage.

File: ui/src/app/panel/panel.component.html [hunk 4]
```diff
@@ -7,81 +7,27 @@
...
     </div>
     <div class="column">
-      <div class="tabs">
+      <!-- placeholder for breadcrumb -->
+      <nav class="breadcrumb" aria-label="breadcrumbs">
         <ul>
-          <li class="is-active"><a>All</a></li>
-          <li><a>New</a></li>
-          <li><a>Popular</a></li>
+          <li><a href="#">Home</a></li>
```
Comment (XuZehua-Teapot) [inline ui/src/app/panel/panel.component.html:25]: What's the purpose of an empty `href`?

File: ui/src/app/search-bar/search-bar.component.ts [hunk 5]
```diff
@@ -48,5 +50,6 @@ export class SearchBarComponent implements OnInit {
   search(): void {
     // for now, only first tag entered is used for search
     this.dataService.fetchThreads(this.tags[0]);
+    this.router.navigate(['/threads']);
```
Comment (XuZehua-Teapot) [inline ui/src/app/search-bar/search-bar.component.ts:53]: (This comment is not in the scope of this PR)
dataService.fetchThreads() won't update until there is a successful response. Should we clear the thread list before anything returns from the backend?

File: ui/src/app/service/data.service.ts (hunk 3 above)
Comment (jybao1) [reply ui/src/app/service/data.service.ts]: Changed to use absolute path in import. I will also fix this for other PRs sent.

File: ui/src/app/search-bar/search-bar.component.ts (hunk 5 above)
Comment (jybao1) [reply ui/src/app/search-bar/search-bar.component.ts:53]: Yes, thread list should be cleared. Thanks for pointing out !

**Pull request merged**